### DISCUSSION TITLE
Custom Library for Extension

### DISCUSSION THREAD

**Discussion by alta-atc** [AntlerVC/firetable#889] [Support / Q&A] [answered]
Hey, Hoping someone has solved this problem previously and can point me in the right direction toward the best approach, since there seem to be a few. 

We are looking to create GCP cloud tasks via a rowy extension,  we need to import a custom library.

`require('@google-cloud/tasks')`

https://cloud.google.com/tasks/docs/creating-http-target-tasks#node.js

**Comment by shamsmosowi** [comment, marked as answer]
hi @alta-atc , you should be able to require it in the `task` type extension (the base extension type, not to be confused gcp tasks)
you might need to add additional [IAM](https://console.cloud.google.com/iam-admin/iam) permissions to the cloud function service account (rowy-backend@`PROJECT_ID`.iam.gserviceaccount.com)

**Reply by alta-atc** [reply]
Thank you. We will take a look at that and let you know how it goes.

**Reply by alta-atc** [reply]
Just wanted to report back in and say that this did indeed work. We also needed to grant IAM permissions to rowy-functions@PROJECT_ID.iam.gserviceaccount.com for creating the cloud tasks within the function. 

This could be a cool idea for an integration, cloud tasks allow for async jobs and are quite useful when dealing with large data sets that generate a lot of changes

Thanks again.
